include/rmm/mr/device/limiting_resource_adaptor.hpp

```diff
@@ -130,11 +132,12 @@ class limiting_resource_adaptor final : public device_memory_resource {
     void* p = nullptr;
 
     std::size_t proposed_size = rmm::detail::align_up(bytes, allocation_alignment_);
-    if (proposed_size + allocated_bytes_ <= allocation_limit_) {
+    allocated_bytes_ += proposed_size;
+    if (allocated_bytes_ <= allocation_limit_) {
       p = upstream_->allocate(bytes, stream);
```

Comment:
what if allocate throws? should we try/catch?

Reply:
Hm, good point. We should probably catch the exception and decrement `allocated_bytes_` and then rethrow.

Reply:
Done.